app/src/main/java/com/bnyro/contacts/ui/models/ContactsModel.kt

```diff
@@ -121,13 +121,27 @@ class ContactsModel : ViewModel() {
         }
     }
 
-    fun getAvailableAccountTypes() = contacts.orEmpty().mapNotNull {
-        it.accountName
-    }.distinct()
+    fun getAvailableAccountTypes(): List<String> {
+        if (contacts.isNullOrEmpty() && contactsHelper is DeviceContactsHelper) return listOf((contactsHelper as DeviceContactsHelper).androidAccountType)
+        return contacts!!.mapNotNull {
+            it.accountType
+        }.distinct()
+    }
+
+    fun getAvailableAccountNames(): List<String> {
+        if (contacts.isNullOrEmpty() && contactsHelper is DeviceContactsHelper) return listOf((contactsHelper as DeviceContactsHelper).deviceContactName)
+        return contacts!!.mapNotNull { it.accountName }.distinct()
+    }
```

Comment:
All that code is being used in `getAvailableAccountTypesAndNames()` too, please just use that existing function and map the items, e.g. 
```kotlin
fun getAvailableAccountNames() = getAvailableAccountTypesAndNames().map { it.second }
fun getAvailableAccountTypes() = getAvailableAccountTypesAndNames().map { it.first }
```